### PR TITLE
remote write 2.0 - follow up improvements

### PR DESCRIPTION
**Pull request by cstyan**
This PR contains various small improvements post metadata PR (https://github.com/prometheus/prometheus/pull/13394). A few more clean up and verification tasks still remain, which I will open issues for.

The list of changes is:
- renaming the field on the scrape config options from `EnableMetadataStorage` to `AppendMetadata`
- move the remote write proto config from a remote write _global_ config to a per remote write config, meaning each remote write config can have a different proto version being used
- various function renaming to clarify that the functions are for proto v2, for example `buildMinimizedWriteRequestStr` -> `buildV2WriteRequest`
- pass through and check/enforce that the metadata is being written to the wal if the selected proto format is 2.0

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1013,6 +1013,9 @@ func CheckTargetAddress(address model.LabelValue) error {
 	return nil
 }
 
+// This needs to live here rather than in the remote package to avoid an import cycle.
+type RemoteWriteFormat int64 //nolint:revive // exported.
```
Comment (bwplotka) [inline config/config.go:1017]:
```suggestion
type RemoteWriteFormat int64
```

File: storage/remote/client.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/prometheus/prometheus/config"
```
Comment (bwplotka) [inline storage/remote/client.go:21]: Looks like `goimports` was not used

File: web/web.go [hunk 3]
```diff
@@ -262,7 +261,8 @@ type Options struct {
 	EnableOTLPWriteReceiver    bool
 	IsAgent                    bool
 	AppName                    string
-	RemoteWriteFormat          remote.RemoteWriteFormat
+	// TODO: should this change to a list of tuples, maybe via the content negotation PR?
```
Comment (bwplotka) [inline web/web.go:264]:
```suggestion
	// TODO(cstyan): Should this change to a list of tuples, maybe via the content negotiation PR?
```

**Review by bwplotka — APPROVED** [review]
Thanks! Minor style nits only.

File: config/config.go [hunk 4]
```diff
@@ -1022,6 +1025,7 @@ type RemoteWriteConfig struct {
 	Name                 string            `yaml:"name,omitempty"`
 	SendExemplars        bool              `yaml:"send_exemplars,omitempty"`
 	SendNativeHistograms bool              `yaml:"send_native_histograms,omitempty"`
+	ProtocolVersion      RemoteWriteFormat `yaml:"remote_write_version,omitempty"`
```
Comment (npazosmendez) [inline config/config.go:1028]: Should we update any docs somewhere to include this?

**Review by npazosmendez — APPROVED** [review]
LGTM, great cleanup

**Review by jesusvazquez — APPROVED** [review]
Changes look good to me 👍

**Pull request merged**